ani-cli

```diff
@@ -134,523 +81,223 @@ update_script () {
-search_history () {
-	[ ! -s "$logfile" ] && die "History is empty"
-	progress "Processing $scrape.."
-	search_results=$(while read -r anime_id; do process_hist_entry & done < "$logfile"; wait)
-	[ -z "$search_results" ] && die "No unwatched episodes"
-	one_hist=$(printf '%s\n' "$search_results" | grep -e "$" -c)
-	[ "$one_hist" = 1 ] && select_first=1
-	anime_selection "$search_results"
-	ep_choice_start=$(sed -n -E "s/${selection_id}\t(.*)/\1/p" "$logfile")
-}
-
-##################
-# URL PROCESSING #
-##################
-
+# generates links based on given provider
 generate_link() {
 	case $1 in
-		2)
-			provider_name='Xstreamcdn'
-			progress "Fetching $provider_name links.."
-			fb_id=$(printf "%s" "$resp" | sed -n "s_.*fembed.*/v/__p")
-			refr="https://fembed-hd.com/v/$fb_id"
-			[ -z "$fb_id" ] && return 0
-			result_links="$(curl -A "$agent" -s -X POST "https://fembed-hd.com/api/source/$fb_id" -H "x-requested-with:XMLHttpRequest" |
-				sed -e 's/\\//g' -e 's/.*data"://' | tr "}" "\n" | sed -nE 's/.*file":"(.*)","label":"(.*)","type.*/\2>\1/p')"
-			;;
-		1)
-			provider_name='Animixplay'
-			progress "Fetching $provider_name Direct link.."
-			refr="$base_url"
-			[ -z "$id" ] && return 0
-			enc_id=$(printf "%s" "$id" | base64)
-			ani_id=$(printf "%sLTXs3GrU8we9O%s" "$id" "$enc_id" | base64)
-			result_links="$(curl -s "$base_url/api/cW9${ani_id}" -A "$agent" -I | sed -nE 's_[L|l]ocation: https?://[^#]*#([^#]*).*_\1_p' | base64 -d)"
-			;;
-		*)
-			provider_name='Gogoanime'
-			progress "Fetching $provider_name Direct link.."
-			refr="$gogohd_url"
-			[ -z "$id" ] && return 0
-			secret_key=$(printf "%s" "$resp" | sed -n '2p' | tr -d "\n" | od -A n -t x1 | tr -d " |\n")
-			iv=$(printf "%s" "$resp" | sed -n '3p' | tr -d "\n" | od -A n -t x1 | tr -d " |\n")
-			second_key=$(printf "%s" "$resp" | sed -n '4p' | tr -d "\n" | od -A n -t x1 | tr -d " |\n")
-			token=$(printf "%s" "$resp" | head -1 | base64 -d | openssl enc -d -aes256 -K "$secret_key" -iv "$iv" | sed -nE 's/.*&(token.*)/\1/p')
-			ajax=$(printf '%s' "$id" | openssl enc -e -aes256 -K "$secret_key" -iv "$iv" -a)
-			data=$(curl -A "$agent" -sL -H "X-Requested-With:XMLHttpRequest" "${gogohd_url}encrypt-ajax.php?id=${ajax}&alias=${id}&${token}" | sed -e 's/{"data":"//' -e 's/"}/\n/' -e 's/\\//g')
-			result_links="$(printf '%s' "$data" | base64 -d 2>/dev/null | openssl enc -d -aes256 -K "$second_key" -iv "$iv" 2>/dev/null |
-				tr -d \\\\ | sed -nE "s_.*file\":\"([^\"]*)\".*source.*_\1_p")"
-			;;
+	1) provider_init 'vrv|wixmp' '/Default :/p' ;; # vrv,wixmp(default)(m3u8)(multi) -> (mp4)(multi)
+	2) provider_init 'pstatic' '/Default B :/p' ;; # pstatic(default backup)(mp4)(multi)
+	3) provider_init 'sharepoint' '/S-mp4 :/p' ;;  # sharepoint(mp4)(single)
+	4) provider_init 'usercloud' '/Uv-mp4 :/p' ;;  # usercloud(mp4)(single)
+	*) provider_init 'gogoanime' '/Luf-mp4 :/p' ;; # gogoanime(m3u8)(multi)
 	esac
+	# logic yet to implement
```

Comment:
What is missing from here? What is yet to be implemented?

Reply:
Range selection ?

Reply:
I doubt that, this is not where it'll be, but yes, it's still not in the script